src/object.rs

```diff
@@ -0,0 +1,334 @@
+        // [9]          (object name, space-padded)
+        // [1]          (object live =  '*' dead = '_')
+        // [7]          (timestamp)
+        // [13]         (compressed position data)
+        // [43]         (comment)
+
+        // packet must have at least name + liveness + timestamp --> (9+1+7) so check that here
+
+        if b.len() < 17 {
+            return Err(DecodeError::InvalidObjectFormat(
+                b.to_vec(),
+                format!(
+                    "packet length must be at least 17 bytes long, got {}",
+                    b.len()
+                ),
+            ));
+        }
+
+        let name = b
+            .get(..9)
+            .map(|by| String::from_utf8_lossy(by).trim().to_string())
+            .unwrap();
+        let live = match *b.get(9).unwrap() as char {
+            '*' => Ok(true),
+            ' ' => Ok(false),
+            others => Err(DecodeError::InvalidObjectFormat(
+                vec![*b.get(9).unwrap()],
+                format!("expected either '*' or '(space)' for object liveness, got '{others}'"),
+            )),
+        }?;
+
+        let timestamp = Timestamp::try_from(b.get(10..17).unwrap())?;
+
+        // ok, here we switch, it COULD be compressed, or it COULD be uncompressed
+        // if the leading character is a numeric, it's an uncompressed
+        // if it's alpha it's compressed
+        let compressed = !(*b.get(17).unwrap_or(&0) as char).is_numeric();
+
+        if compressed {
+            let symbol_table = b[17] as char;
+            let comp_lat = &b[18..22];
+            let comp_lon = &b[22..26];
+            let symbol_code = b[26] as char;
+            let course_speed = &b[27..29];
+            let comp_type = b[29];
+
+            let latitude = Latitude::parse_compressed(comp_lat)?;
+            let longitude = Longitude::parse_compressed(comp_lon)?;
+
+            // From the APRS spec - if the c value is a space,
+            // the csT doesn't matter
+            let cst = match course_speed[0] {
+                b' ' => AprsCst::CompressedNone,
+                _ => {
+                    let t = comp_type
+                        .checked_sub(33)
+                        .ok_or_else(|| DecodeError::InvalidPosition(b.to_owned()))?
+                        .into();
+                    let cs = AprsCompressedCs::parse(course_speed[0], course_speed[1], t)?;
+                    AprsCst::CompressedSome { cs, t }
+                }
+            };
+
+            let comment = b[30..].to_owned();
+
+            Ok(Self {
+                to,
+                name,
+                live,
+                timestamp,
+                latitude,
+                longitude,
+                precision: Precision::default(),
+                symbol_table,
+                symbol_code,
+                comment,
+                cst,
+            })
+        } else {
+            let (latitude, precision) = Latitude::parse_uncompressed(&b[17..25])?;
```

Comment:
Same thing here. Probably could be replaced with `AprsPosition#parse_uncompressed`